src/compiler/scala/tools/nsc/typechecker/RefChecks.scala

```diff
@@ -1451,9 +1446,9 @@ abstract class RefChecks extends Transform {
             )
       }
 
-      sym.isSourceMethod &&
+      sym.name == nme.apply &&
+        !(sym hasFlag STABLE) && // ???
```

Comment:
The best I can come up with is that STABLE is assumed to only be added to
compiler-generated methods. An indirect way of detecting those versus ones
written by the user? (Note that lazy accessors have user written code and
the STABLE flag though)
On Mon, Aug 22, 2016 at 12:10 Lukas Rytz <notifications@github.com> wrote:

> In src/compiler/scala/tools/nsc/typechecker/RefChecks.scala
> <https://github.com/scala/scala/pull/5294#discussion_r75739327>:
>
> > @@ -1451,9 +1446,9 @@ abstract class RefChecks extends Transform {
> >              )
> >        }
> >
> > -      sym.isSourceMethod &&
> > +      sym.name == nme.apply &&
> > +        !(sym hasFlag STABLE) && // ???
>
> hmm.. also wondering about this comment // 🤔🤔🤔
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/scala/scala/pull/5294/files/5d75aae5d391c05105aaadac37aa5f972829041d#r75739327>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAFjy6oz9Njr9qWVBD6vairboFT4l5paks5qifQdgaJpZM4JRUA6>
> .
>